core/src/main/scala/org/apache/spark/util/collection/OpenHashSet.scala

```diff
@@ -114,6 +118,21 @@ class OpenHashSet[@specialized(Long, Int) T: ClassTag](
     rehashIfNeeded(k, grow, move)
   }
 
+  /**
+   * Remove an element from the set. If an element does not exists in the set, nothing is done.
+   */
+  def remove(k: T): Unit = {
```

Comment:
Maybe we should not add `remove` method unless we can add it by a simple way. This is used in many places and this might affect their performance. How about using other implementation?

Reply:
If we need to keep an order without duplication, we can implement this by inroducing another hashset or searching a result array when we try to add an new element.